models/articles.models.js

```diff
@@ -1,15 +1,37 @@
 const db = require("../db/connection");
-const { checkUserExists } = require("../utils/utils.js");
+const { checkUserExists, checkTopicExists } = require("../utils/utils.js");
 
-exports.fetchArticles = () => {
-  const queryString = `
+exports.fetchArticles = (sort_by, order, topic) => {
+  const validColumns = [
+    "article_id",
+    "title",
+    "topic",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
+  if (!validColumns.includes(sort_by)) sort_by = "created_at";
+  if (!["asc", "desc"].includes(order)) order = "desc";
+
+  let queryString = `
     SELECT articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, CAST(COUNT(comments.comment_id) AS INT) AS comment_count 
     FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;
-  `;
-  return db.query(queryString).then((result) => result.rows);
+    LEFT JOIN comments ON articles.article_id = comments.article_id`;
+    const queryStringOrderByGroupBy = ` GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`
+  
+  if (topic === undefined) {
```

Comment:
We should only be using/making **one final query string** to query the DB for this model. This one string is built up dynamically according to the presence (or not) of the queries and then when built used in the only DB query in the model.

A bit of a refactor and some conditional should see you right 👍